explainaboard/loaders/file_loader.py

```diff
@@ -306,12 +484,41 @@ def replace_labels(cls, features: dict, example: dict) -> dict:
                 new_example[examp_k] = examp_v
         return new_example
 
-    def load_raw(self, data: str | DatalabLoaderOption, source: Source) -> list[dict]:
+    def load_raw(
+        self, data: str | DatalabLoaderOption, source: Source
+    ) -> FileLoaderReturn:
         config = narrow(data, DatalabLoaderOption)
         dataset = load_dataset(
             config.dataset, config.subdataset, split=config.split, streaming=False
         )
-        return [self.replace_labels(dataset.info.features, x) for x in dataset]
+        # TODO(gneubig): patch for an inconsistency in datalab, where DatasetDict
+        #  doesn't have info
+        if isinstance(dataset, DatasetDict) or isinstance(dataset, IterableDatasetDict):
+            raise ValueError('Cannot handle DatasetDict returns')
+        info = dataset.info
+        # Infer metadata from the dataset
+        metadata = FileLoaderMetadata()
+        if info.languages is not None:
+            metadata.supported_languages = info.languages
+            if (
+                metadata.supported_languages
+                and len(metadata.supported_languages) > 0
```

Comment:
(1) it would be nice if docstring is added here
(2) what if `len(metadata.supported_languages)` > 3?